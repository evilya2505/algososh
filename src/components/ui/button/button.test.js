import React from "react";
import renderer from "react-test-renderer";
import { Button } from "./button";

const onClickMock = jest.fn();

describe("Button Component", () => {
  it("рендер кнопки с текстом", () => {
    const tree = renderer
      .create(<Button text="Нажми меня" onClick={onClickMock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер кнопки без текста", () => {
    const tree = renderer.create(<Button onClick={onClickMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер заблокированной кнопки", () => {
    const tree = renderer
      .create(
        <Button text="Заблокировано" disabled={true} onClick={onClickMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер кнопки с индикацией загрузки", () => {
    const tree = renderer
      .create(
        <Button text="Загрузка..." isLoader={true} onClick={onClickMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("корректный вызов колбека при клике на кнопку", () => {
    const tree = renderer.create(
      <Button text="Нажми меня" onClick={onClickMock} />
    ).root;

    // Находим кнопку в снэпшоте
    const button = tree.findByType("button");

    // Эмулируем клик на кнопку
    button.props.onClick();

    // Проверяем, что колбек был вызван
    expect(onClickMock).toHaveBeenCalled();
  });
});
