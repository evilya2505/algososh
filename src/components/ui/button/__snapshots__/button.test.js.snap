// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component рендер заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  >
    Заблокировано
  </p>
</button>
`;

exports[`Button Component рендер кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Component рендер кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Component рендер кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;
